rules_gathering/toolchains.bzl

```diff
@@ -0,0 +1,54 @@
+"""
+This module contains the provider for a sbom_tools toolchain as well as a basic implementation
+that invokes binaries directly with well known cli arguments.
+"""
+
+SbomToolsInfo = provider("Toolchain used to generate and maniputate SBOMs", fields = [
+    "create_intermediate_sbom",
+    "merge_intermediate_sboms", 
+    "convert_intermediate_sbom_to_format"
+])
```

Comment:
I would prefer to keep the provider in a separate file from the default implementation.
Some folks have already mentioned that they want to use their implementations written in Go/Java.